server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/TestLoginManager.java

```diff
@@ -173,6 +176,7 @@ public static boolean isOAuthTestLoginRequest(HttpServletRequest request) {
         return "test".equals(parseRegistrationName(registrationId));
     }
 
+    @PreAuthenticate(actions = "read", resourceType = "ODC_INTEGRATION", isForAll = true)
```

Comment:
`loadLdapContext` is executed before login, so it does not need to be annotated with @PreAuthenticate.